packages/deployments/contracts/contracts/nomad-xapps/contracts/connext/ConnextHandler.sol

```diff
@@ -0,0 +1,524 @@
+   * @param _newMaxRouters The new max amount of routers
+   */
+  function setMaxRoutersPerTransfer(uint256 _newMaxRouters) external override onlyOwner {
+    if (_newMaxRouters == 0 || _newMaxRouters == maxRoutersPerTransfer)
+      revert ConnextHandler__setMaxRoutersPerTransfer_invalidMaxRoutersPerTransfer();
+
+    maxRoutersPerTransfer = _newMaxRouters;
+
+    emit MaxRoutersPerTransferUpdated(_newMaxRouters, msg.sender);
+  }
+
+  // ============ External functions ============
+
+  receive() external payable {}
+
+  /**
+   * @notice This is used by anyone to increase a router's available liquidity for a given asset.
+   * @dev The liquidity will be held in the local asset, which is the representation if you
+   * are *not* on the canonical domain, and the canonical asset otherwise.
+   * @param _amount - The amount of liquidity to add for the router
+   * @param _local - The address of the asset you're adding liquidity for. If adding liquidity of the
+   * native asset, routers may use `address(0)` or the wrapped asset
+   * @param _router The router you are adding liquidity on behalf of
+   */
+  function addLiquidityFor(
+    uint256 _amount,
+    address _local,
+    address _router
+  ) external payable override nonReentrant {
+    _addLiquidityForRouter(_amount, _local, _router);
+  }
+
+  /**
+   * @notice This is used by any router to increase their available liquidity for a given asset.
+   * @dev The liquidity will be held in the local asset, which is the representation if you
+   * are *not* on the canonical domain, and the canonical asset otherwise.
+   * @param _amount - The amount of liquidity to add for the router
+   * @param _local - The address of the asset you're adding liquidity for. If adding liquidity of the
+   * native asset, routers may use `address(0)` or the wrapped asset
+   */
+  function addLiquidity(uint256 _amount, address _local) external payable override nonReentrant {
+    _addLiquidityForRouter(_amount, _local, msg.sender);
+  }
+
+  /**
+   * @notice This is used by any router to decrease their available liquidity for a given asset.
+   * @param _amount - The amount of liquidity to remove for the router
+   * @param _local - The address of the asset you're removing liquidity from. If removing liquidity of the
+   * native asset, routers may use `address(0)` or the wrapped asset
+   * @param _to The address that will receive the liquidity being removed
+   */
+  function removeLiquidity(
+    uint256 _amount,
+    address _local,
+    address payable _to
+  ) external override nonReentrant {
+    // transfer to specicfied recipient IF recipient not set
+    address recipient = routerRecipients(msg.sender);
+    recipient = recipient == address(0) ? _to : recipient;
+
+    ConnextUtils.removeLiquidity(_amount, _local, recipient, routerBalances, wrapper);
+  }
+
+  function xcall(XCallArgs calldata _args) external payable override returns (bytes32) {
```

Comment:
where is the function descriptor / docstring? :(

Reply:
can add in later in cleanup PRs

Reply:
added